### PR TITLE
🚨🚨🐛 Source Notion: JSON Schema Updates

### LINKED ISSUE #24560: Source Notion: Add undeclared columns to spec
## What
In this [PR](https://github.com/airbytehq/airbyte/pull/23985), `BasicReadTest.test_read` test in the Connector Acceptance Tests was updated to fail if the connector produces
stream records which contain columns that haven't been declared in the spec.

Notion currently fails the updated test. Its `acceptance-test-config.yml` was edited with the
`fail_on_extra_columns: false` parameter in [this PR](https://github.com/airbytehq/airbyte/pull/24552) in order to avoid this change from making the connector fail CAT.

We want to add the undeclared columns to the spec.

**How this will help**:
- Column selection is currently blocked for connectors that don't declare all columns that they pass. This
  is because turning on column selection will stop these undeclared columns from being sent, when they were
  previously being sent. Doing this allows us to enable column selection for Notion!
- Users will better understand the shape of the data they'll receive, since the columns will match the spec.

## How
The following descriptions of streams that pass undeclared columns come from results of the failed connector acceptance test:

```
columns.txt:./source-notion-2177-2023-03-26T03:19:20.8276807Z The databases stream has the following schema errors:
columns.txt:./source-notion:2178:2023-03-26T03:19:20.8277731Z Additional properties are not allowed ('description' was unexpected)
columns.txt:./source-notion-2837-2023-03-26T03:19:20.8691049Z The pages stream has the following schema errors:
columns.txt:./source-notion:2838:2023-03-26T03:19:20.8691667Z Additional properties are not allowed ('created_by', 'icon', 'last_edited_by' were unexpected)
columns.txt:./source-notion-3266-2023-03-26T03:19:20.8930751Z The blocks stream has the following schema errors:
columns.txt:./source-notion:3267:2023-03-26T03:19:20.8931130Z Additional properties are not allowed ('last_edited_by', 'created_by', 'parent' were unexpected)
columns.txt:./source-notion-4751-2023-03-26T03:21:51.4323870Z The databases stream has the following schema errors:
columns.txt:./source-notion:4752:2023-03-26T03:21:51.4324908Z Additional properties are not allowed ('description' was unexpected)
columns.txt:./source-notion-5411-2023-03-26T03:21:51.4785392Z The pages stream has the following schema errors:
columns.txt:./source-notion:5412:2023-03-26T03:21:51.4786103Z Additional properties are not allowed ('icon', 'last_edited_by', 'created_by' were unexpected)
columns.txt:./source-notion-5840-2023-03-26T03:21:51.5055136Z The blocks stream has the following schema errors:
columns.txt:./source-notion:5841:2023-03-26T03:21:51.5055513Z Additional properties are not allowed ('last_edited_by', 'created_by', 'parent' were unexpected)
columns.txt:./source-notion-7324-2023-03-26T03:24:06.3162835Z The databases stream has the following schema errors:
columns.txt:./source-notion:7325:2023-03-26T03:24:06.3163809Z Additional properties are not allowed ('description' was unexpected)
columns.txt:./source-notion-7984-2023-03-26T03:24:06.3623606Z The pages stream has the following schema errors:
columns.txt:./source-notion:7985:2023-03-26T03:24:06.3624201Z Additional properties are not allowed ('last_edited_by', 'created_by', 'icon' were unexpected)
columns.txt:./source-notion-8413-2023-03-26T03:24:06.3873864Z The blocks stream has the following schema errors:
columns.txt:./source-notion:8414:2023-03-26T03:24:06.3874238Z Additional properties are not allowed ('last_edited_by', 'created_by', 'parent' were unexpected)

```

1. Add the missing properties indicated by the `Additional properties are not allowed ('<column>', 'column' were unexpected)` logs to the connector's spec.
2. Remove `fail_on_extra_columns: false` from the `acceptance-test-config.yml` file.
3. Commit changes to spec and `acceptance-test-config.yml` and open a PR.
4. Run tests on the connector, either automatically via CI or manually via the `/test` command
5. Profit!

Definition of done: Notion passes CAT without declaring `fail_on_extra_columns: false`. If the
API starts sending over extra columns in the future, we will catch and fix them as part of the #connector-health
movement.

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What
Updates to JSON schemas for all existing streams

## How
- Added unexpected columns for all streams
- Added date-time format to `last_edited_time` and `created_time` fields for all streams
- BREAKING CHANGE: Renamed `text` property to `rich_text` in objects that contain rich text. [Notion changelog for this update](https://developers.notion.com/page/changelog#releasing-notion-version-2022-02-22). 
- BREAKING CHANGE: Updated type of `rich_text` property in Pages stream from object to array of `rich_text` objects, which should also resolve [this PR](https://github.com/airbytehq/airbyte/pull/25923).
- BREAKING CHANGE: Updated schema of Rich Text object to reflect API object. [Notion docs](https://developers.notion.com/reference/rich-text)
- Updated schema for `callout`, `code` and `to-do` objects in Blocks stream
- BREAKING CHANGE: Changed type of `phone_number` property in Database stream from string to object

## User Impact
Many of the changes are simply additions of missing fields, but every stream other than Users is affected in some way by removal/renaming of properties. Pages properties `rich_text` and `phone_number` also have a type change.

Customers will have to refresh the schemas and reset data.

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`281e11740f`](https://github.com/airbytehq/airbyte/pull/30587/commits/281e11740fa91e4192ae4d0db6a280ed825ca415)) - ❌

⏲️  Total pipeline duration: 03mn50s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-notion docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-notion/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1695415321/281e11740fa91e4192ae4d0db6a280ed825ca415/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:281e11740fa91e4192ae4d0db6a280ed825ca415)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`d9ee6dbf78`](https://github.com/airbytehq/airbyte/pull/30587/commits/d9ee6dbf783c4de9067df23d97a2be429cf109b5)) - ❌

⏲️  Total pipeline duration: 04mn27s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-notion docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-notion/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1695660688/d9ee6dbf783c4de9067df23d97a2be429cf109b5/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d9ee6dbf783c4de9067df23d97a2be429cf109b5)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`12c6df607e`](https://github.com/airbytehq/airbyte/pull/30587/commits/12c6df607e7ac27f1d64510964d9ac7e319da674)) - ❌

⏲️  Total pipeline duration: 03mn21s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-notion docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-notion/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1695661776/12c6df607e7ac27f1d64510964d9ac7e319da674/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:12c6df607e7ac27f1d64510964d9ac7e319da674)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`6b2b5815e5`](https://github.com/airbytehq/airbyte/pull/30587/commits/6b2b5815e59ad9de7e21e88be2951a1da69821c0)) - ❌

⏲️  Total pipeline duration: 03mn25s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-notion docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-notion/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1695662200/6b2b5815e59ad9de7e21e88be2951a1da69821c0/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:6b2b5815e59ad9de7e21e88be2951a1da69821c0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Review by maxi297 — APPROVED** [review]
Just one small question but given this is address, we're good to go!

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`974c2f2604`](https://github.com/airbytehq/airbyte/pull/30587/commits/974c2f26044ce7328b7a1becebb35b0f2e5907fe)) - ✅

⏲️  Total pipeline duration: 04mn15s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Connector package install                                            | ✅        |
| Build source-notion docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |
| Code format checks                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-notion/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1695677322/974c2f26044ce7328b7a1becebb35b0f2e5907fe/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:974c2f26044ce7328b7a1becebb35b0f2e5907fe)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

File: docs/integrations/sources/notion.md [hunk 1]
```diff
@@ -107,7 +107,8 @@ The connector is restricted by Notion [request limits](https://developers.notion
 
 | Version | Date       | Pull Request                                             | Subject                                                                      |
 | :------ | :--------- | :------------------------------------------------------- | :--------------------------------------------------------------------------- |
-| 1.1.2   | 2023-08-30 | [29999](https://github.com/airbytehq/airbyte/pull/29999) | Update error handling during connection check
+| 1.2.0   | 2023-09-19 | [30587](https://github.com/airbytehq/airbyte/pull/30587) | Source-wide schema update                                                    |
```
[the remark this answers is not in the document]
Comment (ChristoGrab) [reply docs/integrations/sources/notion.md]: Oh whoops, thanks for catching that! Fixed 👍

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`dae994798e`](https://github.com/airbytehq/airbyte/pull/30587/commits/dae994798e1ac8fdc582b16a6e4840982287350a)) - ✅

⏲️  Total pipeline duration: 03mn34s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Connector package install                                     | ✅        |
| Build source-notion docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ✅        |
| Code format checks                                            | ✅        |
| Validate metadata for source-notion                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1696436818/dae994798e1ac8fdc582b16a6e4840982287350a/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:dae994798e1ac8fdc582b16a6e4840982287350a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`db33a1ac47`](https://github.com/airbytehq/airbyte/pull/30587/commits/db33a1ac471752ed2a90876337216a9354e818ed)) - ✅

⏲️  Total pipeline duration: 01mn20s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Connector package install                                     | ✅        |
| Build source-notion docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ✅        |
| Code format checks                                            | ✅        |
| Validate metadata for source-notion                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1696437583/db33a1ac471752ed2a90876337216a9354e818ed/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:db33a1ac471752ed2a90876337216a9354e818ed)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`4c71f01e1b`](https://github.com/airbytehq/airbyte/pull/30587/commits/4c71f01e1bf32bba1f8f15b4822bb9baecc8e4a2)) - ❌

⏲️  Total pipeline duration: 04mn30s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Connector package install                                     | ✅        |
| Build source-notion docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ❌        |
| Code format checks                                            | ✅        |
| Validate metadata for source-notion                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1696867532/4c71f01e1bf32bba1f8f15b4822bb9baecc8e4a2/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:4c71f01e1bf32bba1f8f15b4822bb9baecc8e4a2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`54d566e864`](https://github.com/airbytehq/airbyte/pull/30587/commits/54d566e864475e4781a15f6dffc482524d80467f)) - ❌

⏲️  Total pipeline duration: 03mn45s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Connector package install                                     | ✅        |
| Build source-notion docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ❌        |
| Code format checks                                            | ✅        |
| Validate metadata for source-notion                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1696868540/54d566e864475e4781a15f6dffc482524d80467f/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:54d566e864475e4781a15f6dffc482524d80467f)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`d41e6423e2`](https://github.com/airbytehq/airbyte/pull/30587/commits/d41e6423e2b2ee8c35ec69bd220ac70b987723a0)) - ✅

⏲️  Total pipeline duration: 03mn40s 

| Step                                                          | Result   |
|:--------------------------------------------------------------|:---------|
| Connector package install                                     | ✅        |
| Build source-notion docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                    | ✅        |
| Acceptance tests                                              | ✅        |
| Code format checks                                            | ✅        |
| Validate metadata for source-notion                           | ✅        |
| Connector version semver check                                | ✅        |
| Connector version increment check                             | ✅        |
| QA checks                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_fix-notion-schemas/1696871560/d41e6423e2b2ee8c35ec69bd220ac70b987723a0/source-notion/2.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d41e6423e2b2ee8c35ec69bd220ac70b987723a0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Pull request merged**